### ISSUE TITLE
Support .desktop files

### ISSUE THREAD

**Issue by Enn3Developer** [axodotdev/cargo-dist#1489]
It'd be nice if one can configure the shell installer to create a .desktop file on Linux, especially for GUI applications

**Comment by ashleygwilliams** [comment]
hey @Enn3Developer! that's a cool request- while it's not currently something we can prioritize, we would totally accept a PR for it- and are happy to help point you in the right direction either here or on our discord (https://discord.gg/XAFG6xSZ). 

are you shipping some GUI applications? always love to hear what people are shipping with dist!

**Comment by Enn3Developer** [comment]
Yes, it's a music player and I found that dist simplifies my release workflow.
I already joined the discord so we could continue there

**Comment by duckinator** [comment]
I think it makes sense to leverage XDG here, since we're already looking to do so in #287 and it provides a well-defined path for user-installed .desktop files.

- `$XDG_DATA_HOME` defaults to `$HOME/.local/share`.
- .desktop files would go under `$XDG_DATA_HOME/applications`.

The main catch here is .desktop files usually reference icons. As best I can tell, the most portable way is to provide an icon yourself and specify an absolute path to it. The [freedesktop.org desktop entry spec](https://specifications.freedesktop.org/desktop-entry-spec/latest/value-types.html) implies that you can't use e.g. an inlined base64-encoded image file, or at least can't guarantee that'll work. (Which is rather unfortunate.)

**Comment by musjj** [comment]
Icons can simply be placed in `~/.local/share/icons` and referenced in the desktop file. At least this is how it works in most systems I've used.
